components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/ConfigsResource.java

```diff
@@ -0,0 +1,174 @@
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.micro.integrator.management.apis;
+
+import com.google.gson.JsonObject;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.commons.json.JsonUtil;
+import org.apache.synapse.config.SynapseConfiguration;
+import org.apache.synapse.transport.passthru.config.PassThroughCorrelationConfigDataHolder;
+import org.json.JSONObject;
+import org.wso2.micro.integrator.management.apis.security.handler.SecurityUtils;
+import org.wso2.micro.integrator.security.user.api.UserStoreException;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.rmi.CORBA.Util;
+
+/**
+ * This resource will handle requests coming to configs/.
+ */
+public class ConfigsResource implements MiApiResource {
+
+    private static final Log LOG = LogFactory.getLog(ConfigsResource.class);
+
+    private static final String CONFIG_NAME = "configName";
+    private static final String CONFIGS = "configs";
+    private static final String CORRELATION = "correlation";
+    private static final String ENABLED = "enabled";
+    Set<String> methods;
+
+    public ConfigsResource() {
+
+        methods = new HashSet<>();
+        methods.add(Constants.HTTP_GET);
+        methods.add(Constants.HTTP_PUT);
+    }
+
+    @Override
+    public Set<String> getMethods() {
+
+        return methods;
+    }
+
+    @Override
+    public boolean invoke(MessageContext messageContext,
+                          org.apache.axis2.context.MessageContext axis2MessageContext, SynapseConfiguration synapseConfiguration) {
+
+        String httpMethod = axis2MessageContext.getProperty(Constants.HTTP_METHOD_PROPERTY) != null ?
+                axis2MessageContext.getProperty(Constants.HTTP_METHOD_PROPERTY).toString() :
+                "method undefined";
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Handling" + httpMethod + "request");
+        }
+        String performedBy = Constants.ANONYMOUS_USER;
+        if (messageContext.getProperty(Constants.USERNAME_PROPERTY) !=  null) {
+            performedBy = messageContext.getProperty(Constants.USERNAME_PROPERTY).toString();
+        }
+        JSONObject response;
+        try {
+            boolean isAdmin = SecurityUtils.isAdmin(performedBy);
```

Comment:
Instead of checking admin permissions here, we can add this config resource as a admin privileged resource in [1]
[1]-https://github.com/wso2/micro-integrator/blob/master/components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/AuthorizationHandler.java#L50